internal/godoc/parse_overload_func.go

```diff
@@ -0,0 +1,79 @@
+package godoc
+
+import (
+	"go/ast"
+	"go/doc"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+// FindOverloadFuncThenAdd  Find overloaded functions and update function names
+func FindOverloadFuncThenAdd(d *doc.Package) {
+	if len(d.Funcs) == 0 {
+		return
+	}
+	var overloadFuncName = make(map[string]string)
+	for _, constO := range d.Consts {
+		for _, name := range constO.Names {
+			if strings.Contains(name, "Gopo_") {
+				for _, spec := range constO.Decl.Specs {
+					if vs, ok := spec.(*ast.ValueSpec); ok {
+						for _, name := range vs.Names {
+							if strings.Contains(name.Name, "Gopo_") {
+								n := strings.TrimPrefix(name.Name, "Gopo_")
+								for _, v := range vs.Values {
+									if bas, ok := v.(*ast.BasicLit); ok {
+										overloadFuncs := strings.Split(bas.Value, ",")
+										for i2 := range overloadFuncs {
+											overloadFuncName[strings.ReplaceAll(overloadFuncs[i2], "\"", "")] = n
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	var overloadFunc = make([]*doc.Func, 0)
+	for _, funcO := range d.Funcs {
+		RestoreName(funcO)
```

Comment:
I think `RestoreName(func0)` should after the following `if expr` because it may refer by `overloadFuncName`.